docs/plugin_developers_guide/02_writing_plugins.rst

```diff
@@ -81,6 +82,26 @@ These are the properties returned by the JSON document:
       }
   }
 
+
+The various values that may appear in the *mode* item are shown in the table below
+
++---------+---------------------------------------------------------------------------------------+
+| Mode    | Description                                                                           |
++=========+=======================================================================================+
+| poll    | The plugin is a polled plugin and *plugin_poll* will be called periodically to obtain |
+|         | new values.                                                                           |
++---------+---------------------------------------------------------------------------------------+
+| async   | The plugin is an asynchronous plugin, *plugin_poll* will not be called and the        |
+|         | plugin will be supplied with a callback function that it calls each time it has a     |
+|         | new value to pass to the system. The *plugin_register_ingest* entry point will be     |
+|         | called to register the callback with the plugin.                                      |
```

Comment:
*plugin_start* is also required for Python async plugins. We should add brief note about it.